DatadogCore/Sources/Core/DatadogCore.swift

```diff
@@ -491,3 +491,12 @@ extension DatadogCore: Flushable {
         }
     }
 }
+
+#if SPM_BUILD
+import DatadogPrivate
+#endif
+
+internal let registerObjcExceptionHandlerOnce: () -> Void = {
+    ObjcException.rethrow = __dd_private_ObjcExceptionHandler.rethrow
```

Comment:
/question What I understand is that `ObjcException.rethrow` is set internally and not intended for customers to use, but is it safe? Is there a way to validate what is assigned to `ObjcException.rethrow` to ensure it is an Objective-C compatible function? It might over-complicate the current code for a minimal gain, but I'm curious.

Reply:
Good question! The `ObjcException.rethrow` is explicitly internal as part of the `DatadogInternal` module, if a customer provides a non objc-try-catch implementation it will be at their own risk but it won't affect the SDK behaviour. So I don't see the don't need for compilation or runtime validation, if any existed.